### ISSUE TITLE
Uyuni Enhancement Request - Uyuni 2021-08 - Replace joda-time

### ISSUE THREAD

**Issue by sbluhm** [uyuni-project/uyuni#4123]
# Short description
spacewalk-java requires package joda-time

Joda-time is no longer in active development except to keep timezone data up to date. From Java SE 8 onwards, users are asked to migrate to java.time (JSR-310) - a core part of the JDK which replaces this project. For Android users, java.time is added in API 26+. Projects needing to support lower API levels can use the ThreeTenABP library.

Removing the joda-time dependency would also remove the joda-convert dependency, saving two additional packages.

**Comment by sbluhm** [comment]
After initial analysis, I don't think it is even used anymore:
```
java $ git grep --name-only joda
buildconf/build-props.xml
buildconf/ivy/ivy-suse.xml
buildconf/ivy/obs-maven-config.yaml
code/src/com/suse/manager/reactor/utils/test/productdata/repositories.json
spacewalk-java.spec
```

**Comment by mcalmer** [comment]
CC @mbologna

**Comment by mbologna** [comment]
> spacewalk-java uses onelogin's SAML Java Toolkit 2.4 which requires JodaTime (package java-saml).
> 
> JodaTime is technically not required anymore and there is an issue opened to remove it:
> [onelogin/java-saml#201](https://github.com/onelogin/java-saml/issues/201)
> Also this stab at removing it:
> [onelogin/java-saml@00dc815](https://github.com/onelogin/java-saml/commit/00dc8155e2ef54b02d39d6c70440d83eec29b04b)
> 
>     * [ ]  First step should be adding the dependencies correctly to java-saml and removing them from spacewalk-java.
> 
>     * [ ]  Get a new SAML release sorted and packaged without JodaTime

Agreed with the steps outlined, not sure if we should already start doing the first while we wait for the second or doing everything as soon as the new release is there.

**Comment by sbluhm** [comment]
Change has been merged into java-saml master with https://github.com/onelogin/java-saml/pull/374 .
Next step would be waiting for the next release.

**Comment by admd** [comment]
Java-saml still use joda, after the commit, there was no more release for. We cannot get rid of it for now.
